entry_types/scrolled/package/src/frontend/MediaPlayer/index.js

```diff
@@ -30,11 +31,13 @@ function PreparedMediaPlayer(props){
   let playerRef = useRef();
   let previousPlayerState = useRef(props.playerState);
   let scrollToSection = useContext(ScrollToSectionContext);
+  let mediaContext = useMediaContext();
 
   let onSetup = (newPlayer)=>{
     playerRef.current = newPlayer;
     newPlayer.on('ended', () => props.nextSectionOnEnd && scrollToSection('next'));
-
+    newPlayer.context = mediaContext;
```

Comment:
To get around having to use such a property as a way to pass data to the context function, we could use the same approach we used with `updateHooks`:

`mediaEvents` could define an `updateMediaEventsContextData` method, `getPlayer` could take an `mediaEventsContextData` option and call that method. 

Then the above `MediaPlayer` test could test that the correct context data is passed to `getPlayer`. I think this would be a clearer interface and separation of concerns. We could then also add a test for `mediaEvents` that makes sure media events include the updated context when `updateMediaEventsContextData` is called.

Reply:
I am not sure what should be the structure of event context object.
From our discussion in some previous tickets, we decided that this will be different in scrolled entries.
also there is no page in scrolled so this confuses me.
Can you please suggest the shape of required context object

Reply:
It's the one I linked in the summary of the review. Scrolled already dispatches `page:change` events to trigger analytics: The object passed there has the desired shape:

https://github.com/codevise/pageflow/blob/9450a8d481e62aa34535794c7b9f5b8065e79edf/entry_types/scrolled/package/src/frontend/useSectionChangeEvents.js#L13-L15

The same sort of object is expected as `context.page` in media events [here](https://gitlab.codevise.de/pageflow/pageflow-analytics/blob/1781c2549270074ddedf37c4e8d77b40e2478b22/app/assets/javascripts/pageflow/analytics/webtrekk/provider.js#L108).

Reply:
So this is what it is I think

`{
  configuration: {
    title: <chapter-title>
  },
  index: <section-index>
  chapterIndex: <>
}`

currently title is going as undefined. let me know if it is required

Reply:
Yes, since we want to send the same data in `page:change` events as in media event context. And as I said above, it would be nice to extract this logic to construct this dummy page object to a shared function in Scrolled.

Reply:
Currently it is being constructed in useMediaEventsContext. Or you mean something else?